### PR TITLE
feat: add hookArg support to createModel

### PR DESCRIPTION
**Pull request by h-a-n-a**
We could pass a second parameter to createModel to set args for the custom hook.

### CODE REVIEW

**Comment by h-a-n-a** [comment]
Related to #24

**Comment by awmleer** [comment]
Good job! But should the second param of `createModel` be named as `hookProps` ? A little bit confusing since `props` is related with React components and is in the form of object.
What's more, the `props` of a component is dynamic, which means when a prop changes, the component should trigger re-rendering. But the `hookProps` here is actually a static value used and should not be changed after the model is created.

Personally, I prefer `param` or `initial`.

cc @brickspert @tli4

**Comment by h-a-n-a** [comment]
> Good job! But should the second param of `createModel` be named as `hookProps` ? A little bit confusing since `props` is related with React components and is in the form of object.
> What's more, the `props` of a component is dynamic, which means when a prop changes, the component should trigger re-rendering. But the `hookProps` here is actually a static value used and should not be changed after the model is created.
> 
> Personally, I prefer `param` or `initial`.
> 
> cc @brickspert @tli4

I figured. Personally, I would take `param`, cause `initial` sounds like a param which will be passed to the custom hook only on its first call. Maybe `arg` is also a good idea.

**Comment by brickspert** [comment]
Maybe you should add test case for this change.

File: README-cn.md [hunk 1]
```diff
@@ -77,6 +77,31 @@ function App(props) {
 
 ## 进阶用法
 
+### 为 custom hook 提供参数
+
+当一个 custom hook 被用于不同的场景下，我们希望它们可以拥有不同的参数。
+
+如下方的例子一样，我们可以使用 `createModel` 的第二个参数，为 custom hook 设置 Props。这是设置初始值的最佳时机。
```
Comment (awmleer) [inline README-cn.md:84]: Documentation needs updating~

File: README.md [hunk 2]
```diff
@@ -77,6 +77,31 @@ function App(props) {
 
 ## Advanced Usages
 
+### Pass props to custom hooks
+
+In some scenarios, we might want to pass props to custom hooks.
+
+Just like the example below, we could pass props to the second parameter of createModel to set props for the custom hook. This is the best time to set the initial value.
```
Comment (awmleer) [inline README.md:84]: Same here.

**Comment by awmleer** [comment]
@h-a-n-a Could you please add some test cases for this new feature?

**Comment by h-a-n-a** [comment]
> @h-a-n-a Could you please add some test cases for this new feature?

Sure!

File: src/__tests__/main.test.tsx [hunk 3]
```diff
@@ -4,14 +4,14 @@ import { Component, FC, memo, useState } from "react";
 import * as testing from "@testing-library/react";
 
 test("simple", function() {
```
Comment (awmleer) [inline src/__tests__/main.test.tsx:6]: It's better to create a new test case instead of modifying this one.

**Comment by awmleer** [comment]
@h-a-n-a Thanks for your great work!

**Pull request merged**